com.jetbrains.IntelliJ-IDEA-Community.json

```diff
@@ -4,22 +4,33 @@
     "runtime": "org.freedesktop.Sdk",
     "runtime-version": "20.08",
     "sdk": "org.freedesktop.Sdk",
+    "sdk-extensions": ["org.freedesktop.Sdk.Extension.openjdk11"],
     "separate-locales": false,
     "finish-args": [
         "--device=dri",
-        "--filesystem=host",
-        "--filesystem=xdg-run/keyring",
         "--share=ipc",
         "--share=network",
         "--socket=ssh-auth",
-        "--socket=wayland",
         "--socket=x11",
+	"--filesystem=~/IdeaProjects:create",
+        "--filesystem=xdg-cache/tmp:create",
+        "--filesystem=xdg-cache/log:create",
+        "--filesystem=xdg-data/plugins:create",
```

Comment:
IntelliJ is already using the XDG Desktop Specification. In the end, it's going to be similar to what you have here. Also, they create different "umbrella" sub-directories for major versions: how is that resolved with this config, provided there is such upgrade?

Reply:
By default everything is in your home, which was the thing that I tried to solve here. The idea was to have all IntelliJ files in /var/data or /var/config and so on, instead of hidden directories in the user home directory. One of the reasons I like flatpak is because it creates a sandboxed environment, so I don't like the filesystem host permission, that I don't use on my machine. 
I used IntelliJ properties described here https://www.jetbrains.com/help/idea/tuning-the-ide.html#default-dirs to reach the goal. The properties are set in the idea.properties file.
Without those settings, IntelliJ will create the files in the home directory and if it doesn't have permissions, they will be deleted when exiting the app.

Reply:
What I was trying to say is that all of that is meant to be there exactly, according to the XDG Base Directory Specification. It wasn't like that until recently. Right now, everything should be under `~/.var/app/com.jetbrains.IntelliJ-IDEA-Community/{xdg-cache, xdg-data, etc.}`.

What I'm trying to understand is why this custom solution is better than the defaults? I somehow understand the `"--persist=~/IdeaProjects:create"` but keep in mind you are forcing everyone to follow that (non-standard) path.

> I'm aware that IntelliJ IDEA uses it by default, but I'm sure not everyone places the projects there.

Reply:
You're right, with the latest versions of IntelliJ, the files are stored in the correct folders, so this step is not needed anymore. I'll remove this part.
Regarding IdeaProjects folder, I understand that not everybody use this convention, I also don't place my projects there, but it's pretty easy to override the filesystem permissions to give IntelliJ's flatpak the access to another folder.
If you see recent Firefox flatpak, it has access only to xdg-downloads and .mozilla folder. I think we should strive to have flatpaks with filesystem permissions to as minimum as possible.

Reply:
OK, I didn't think about that part; so this one could be the defaults (that probably a lot of users are fine with) and I know there is a GUI (along with the CLI) to override permissions — that can be used to customize it and select a different path.

Reply:
I think it's better to have the default location; that's what I would expect. And these sort of things is what I was referring to: we are going to allow each one for every plugin/build tool, etc. But for the Community edition shouldn't be that hard.

Starting with `--persist=~/.m2`, `--persist=~/.gradle` and `--persist=~/.npm` is a good start. I wish all those tools respect the XDG Base Directory Specification, then nothing like this would be necessary. I filed an issue for Maven and Gradle in this regard.

Reply:
I added those permissions. Good idea to open an issue for Maven and Gradle, although I don't see this to change in the recent future.

Reply:
I just want to add that ideally we shouldn't use .m2, .gradle, and other folders that are part of the host configuration files. Indeed, in flatpak documentation it is stated: "Retaining and sharing configuration with non-Flatpak installations is to be avoided." (https://docs.flatpak.org/en/latest/sandbox-permissions.html#filesystem-access). And configuration refers also to those local artifacts repositories IMHO.

For now I agree that we can use those directories but ideally this is not the best option.